scripts/merge-rules.py

```diff
@@ -0,0 +1,42 @@
+#!/usr/bin/env python3
+import os
+import sys
+from ruamel.yaml import YAML
+
+
+def getRulefilesFromPath(path):
+    rulefiles = []
+    for root, dirs, files in os.walk(path):
+        dirs[:] = [d for d in dirs if not d.startswith(".")]
+        for name in files:
+            file = os.path.join(root, name)
+            if (file.endswith(".yml") or file.endswith(".yaml")) and (
+                not name.startswith(".")
+            ):
+                rulefiles.append(file)
+    print(f"Found {len(rulefiles)} rulefiles")
+    return rulefiles
+
+
+def mergeRules(rulefilelist, outputfile):
+    yaml = YAML(typ="rt")
+    rulefile = {"rules": []}
+    for file in rulefilelist:
+        rulefileyaml = yaml.load(open(file))
+        rulefile["rules"] += rulefileyaml["rules"]
+    rulecount = len(rulefile["rules"])
+    print(f"Created {rulecount} rules")
+    output = open(outputfile, "w")
+    yaml.dump(rulefile, output)
+    output.close()
+
+
+if __name__ == "__main__":
+    if len(sys.argv) != 3:
+        print("Merges all yaml/yml files in a directory and all subdirs")
+        print("Skips directories and files prefixed with a .")
+        print(f"Usage {sys.argv[0]} [input folder] [output filename]")
+    else:
```

Comment:
hey @tkisason thanks for the PR! Can we add in a case for no args to print the help? seems to me that `sys.argv[1]` will fail if no args given

Reply:
This is exactly how it works. It will check if there are exactly 3 arguments (script name, input folder and output filename) and this is not satisfied it will print out the help:

```bash
(semgrep-rule-merger) ~/workspace/semgrep-rule-merger> ./semgrep-rule-merger.py
Merges all yaml/yml files in a directory and all subdirs
Skips directories and files prefixed with a .
Usage ./semgrep-rule-merger.py [input folder] [output filename]
(semgrep-rule-merger) ~/workspace/semgrep-rule-merger> ./semgrep-rule-merger.py semgrep-rules
Merges all yaml/yml files in a directory and all subdirs
Skips directories and files prefixed with a .
Usage ./semgrep-rule-merger.py [input folder] [output filename]
(semgrep-rule-merger) ~/workspace/semgrep-rule-merger> ./semgrep-rule-merger.py semgrep-rules out.yml foo
Merges all yaml/yml files in a directory and all subdirs
Skips directories and files prefixed with a .
Usage ./semgrep-rule-merger.py [input folder] [output filename]
```

If 3 arguments are passed, then it will run the needed functions. 

Is this ok or should i change it in some way?

Reply:
@DrewDennison Note the `if len(sys.argv) == 3`... are we good to merge this one?

Reply:
lgtm